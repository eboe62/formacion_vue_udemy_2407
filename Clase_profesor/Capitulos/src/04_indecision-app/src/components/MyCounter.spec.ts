import { describe, test } from 'vitest';
import { mount } from '@vue/test-utils';
import MyCounter from '../../src/components/MyCounter.vue';

describe('<MyCounter />', () => {
    test('should match snapshot', () => {
        const wrapper = mount(MyCounter, {
            props: {
                value: 5,
            },
        });
        expect(wrapper.html()).toMatchSnapshot;
    });

    test('renders the counter value correctly', () => {
        const value = 5;
        const wrapper = mount(MyCounter, {
            props: {
                value: value,
            },
        });
        const [counterLabel, squareLabel] = wrapper.findAll('h3');

        expect(wrapper.find('h3').text()).toContain(`Counter: ${value}`);
        expect(wrapper.find(`[data-testid="square-label"]`).text()).toContain(`Square: ${value * value}`,
        );
        expect(counterLabel.text()).toContain(`Counter: ${value}`);
        expect(squareLabel.text()).toContain(`Square: ${value * value}`);
    });

    test('increments the counter when +1 button is clicked', async () => {
        const value = 5;
        const wrapper = mount(MyCounter, {
            props: {
                value: value,
            },
        });
        const [counterLabel, squareLabel] = wrapper.findAll('h3');

        const [, btnDecrement] = wrapper.findAll('button');
        await btnDecrement.trigger('click');
        await btnDecrement.trigger('click');

        expect(counterLabel.text()).toContain(`Counter: ${value - 2}`);
        expect(squareLabel.text()).toContain(`Square: ${(value - 2) * (value - 2)}`);

    });
});