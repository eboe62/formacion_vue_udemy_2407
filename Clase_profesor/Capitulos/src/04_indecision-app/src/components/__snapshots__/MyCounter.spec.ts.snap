// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MyCounter /> > should match snapshot 1`] = `
"<section>
  <h2>Counter: 10</h2>
  <h3>Square: 100</h3>
  <div><button class="btn">+1</button><button class="btn">-1</button></div>
</section>"
`;
